storage/integration_test.go

```diff
@@ -1377,6 +1376,115 @@ func TestIntegration_ObjectIterationMatchGlob(t *testing.T) {
 	})
 }
 
+func TestIntegration_ObjectIterationManagedFolder(t *testing.T) {
+	ctx := skipGRPC("not yet implemented in gRPC")
+	multiTransportTest(skipJSONReads(ctx, "no reads in test"), t, func(t *testing.T, ctx context.Context, _ string, prefix string, client *Client) {
+		newBucketName := prefix + uidSpace.New()
+		h := testHelper{t}
+		bkt := client.Bucket(newBucketName).Retryer(WithPolicy(RetryAlways))
+
+		h.mustCreate(bkt, testutil.ProjID(), nil)
+
+		// Enable UBLA as this is necessary for managed folders.
+		if _, err := bkt.Update(ctx, BucketAttrsToUpdate{
+			UniformBucketLevelAccess: &UniformBucketLevelAccess{
+				Enabled: true,
+			},
+		}); err != nil {
+			t.Fatalf("setting UBLA on bucket %v: %v", newBucketName, err)
+		}
+
+		t.Cleanup(func() {
+			if err := killBucket(ctx, client, newBucketName); err != nil {
+				log.Printf("deleting %q: %v", newBucketName, err)
+			}
+		})
+		const defaultType = "text/plain"
+
+		// Populate object names and make a map for their contents.
+		objects := []string{
+			"obj1",
+			"obj2",
+			"obj/with/slashes",
+			"obj/",
+			"other/obj1",
+		}
+		contents := make(map[string][]byte)
+
+		// Test Writer.
+		for _, obj := range objects {
+			c := randomContents()
+			if err := writeObject(ctx, bkt.Object(obj), defaultType, c); err != nil {
+				t.Errorf("Write for %v failed with %v", obj, err)
+			}
+			contents[obj] = c
+		}
+
+		// Create a managed folder. This requires using the Apiary client as this is not available
+		// in the veneer layer.
+		call := client.raw.ManagedFolders.Insert(newBucketName, &raw.ManagedFolder{Name: "mf"})
+		mf, err := call.Context(ctx).Do()
+		if err != nil {
+			t.Fatalf("creating managed folder: %v", err)
+		}
+
+		t.Cleanup(func() {
+			call := client.raw.ManagedFolders.Delete(newBucketName, mf.Name)
+			call.Context(ctx).Do()
```

Comment:
Do we want to catch the error here? I ask because while I know other buckets that don't get properly cleaned up for whatever reason are killed when re-running the tests, but not sure if it would work on a bucket that contains managed folders. Or maybe managed folders should be added to the killBucket function?

Reply:
I would do this but I don't think it's workable right now because killBucket must work for gRPC clients as well. Will add a to-do.